programs/server/config.xml

```diff
@@ -66,69 +66,289 @@
+          -->
+        <!--
+        <secure_native_tcp>
+            <protocol>native_tcp</protocol>
+            <port>9440</port>
+            <enable_tls>yes</enable_tls>
+        </secure_native_tcp>
+        -->
+
+        <!-- Interface for interactions using plain native HTTP API.
+             This interface is also used by ODBC and JDBC drivers (DataGrip, Dbeaver, ...)
+             and by most of web interfaces (embedded UI, Grafana, Redash, ...).
+             See also 'secure_native_http' for secure connections.
+             All parameters relevant to a TCP-based protocol can be specified here. See 'plain_native_tcp'.
+          -->
+        <plain_native_http>
+            <protocol>native_http</protocol>
+            <port>8123</port>
+            <enable_tls>no</enable_tls>
+        </plain_native_http>
+
+        <!-- Interface for interactions using secure native HTTPS API.
+             You have to configure certificate to enable this interface.
+             See the openSSL section below.
+             See also 'plain_native_http' for insecure connections.
+             All parameters relevant to a HTTP-based protocol can be specified here. See 'plain_native_http'.
+          -->
+        <!--
+        <secure_native_http>
+            <protocol>native_http</protocol>
+            <port>8443</port>
+            <enable_tls>yes</enable_tls>
+        </secure_native_http>
+        -->
+
+        <!-- Interface for interactions using plain (or secure, if 'grpc.enable_ssl' is set to 'true') native gRPC protocol.
+             See also 'grpc' for further configutarion.
+             Some parameters relevant to a TCP-based protocol ('listen', 'port', 'listen_try') can be specified here. See 'plain_native_tcp'.
+          -->
+        <!--
+        <plain_native_grpc>
+            <protocol>native_grpc</protocol>
+            <port>9100</port>
+        </plain_native_grpc>
+        -->
+
+        <!-- Interface for interactions using MySQL compatibility protocol.
+             ClickHouse will pretend to be MySQL for applications connecting to this port.
+             All parameters relevant to a TCP-based protocol can be specified here. See 'plain_native_tcp'.
+          -->
+        <plain_mysql>
+            <protocol>mysql</protocol>
+            <port>9004</port>
+            <enable_tls>no</enable_tls> <!-- TLS connections are not supported. -->
+        </plain_mysql>
+
+        <!-- Interface for interactions using PostgreSQL compatibility protocol.
+             ClickHouse will pretend to be PostgreSQL for applications connecting to this port.
+             All parameters relevant to a TCP-based protocol can be specified here. See 'plain_native_tcp'.
+          -->
+        <plain_postgresql>
+            <protocol>postgresql</protocol>
+            <port>9005</port>
+            <enable_tls>no</enable_tls> <!-- TLS connections are not supported. -->
```

Comment:
https://www.postgresql.org/docs/9.1/ssl-tcp.html

Reply:
See reasoning for MySQL.